internal/mode/static/telemetry/collector.go

```diff
@@ -167,23 +169,31 @@ func collectGraphResourceCount(
 	return ngfResourceCounts, nil
 }
 
-func collectNGFReplicaCount(ctx context.Context, k8sClient client.Reader, podNSName types.NamespacedName) (int, error) {
+func collectNGFReplicaData(
+	ctx context.Context,
+	k8sClient client.Reader,
+	podNSName types.NamespacedName,
+) (int, string, error) {
 	var pod v1.Pod
 	if err := k8sClient.Get(
 		ctx,
 		types.NamespacedName{Namespace: podNSName.Namespace, Name: podNSName.Name},
 		&pod,
 	); err != nil {
-		return 0, fmt.Errorf("failed to get NGF Pod: %w", err)
+		return 0, "", fmt.Errorf("failed to get NGF Pod: %w", err)
 	}
 
 	podOwnerRefs := pod.GetOwnerReferences()
 	if len(podOwnerRefs) != 1 {
-		return 0, fmt.Errorf("expected one owner reference of the NGF Pod, got %d", len(podOwnerRefs))
+		return 0, "", fmt.Errorf("expected one owner reference of the NGF Pod, got %d", len(podOwnerRefs))
 	}
 
 	if podOwnerRefs[0].Kind != "ReplicaSet" {
-		return 0, fmt.Errorf("expected pod owner reference to be ReplicaSet, got %s", podOwnerRefs[0].Kind)
+		return 0, "", fmt.Errorf("expected pod owner reference to be ReplicaSet, got %s", podOwnerRefs[0].Kind)
+	}
+
+	if podOwnerRefs[0].UID == "" {
+		return 0, "", fmt.Errorf("expected pod owner reference to have a UID: %v", podOwnerRefs[0])
```

Comment:
nit: 
```suggestion
		return 0, "", fmt.Errorf("expected pod owner reference to have a UID")
```

No need to print the empty string.

Reply:
prints the data available about the `ownerReference` of the pod which contains information like name, type..

Reply:
I see, my bad